docs/source/guides/advanced_custom_primitives.ipynb

```diff
@@ -0,0 +1,233 @@
+   "outputs": [],
+   "source": [
+    "class StringCount(TransformPrimitive):\n",
+    "    '''Count the number of times the string value occurs.'''\n",
+    "    name = 'string_count'\n",
+    "    input_types = [NaturalLanguage]\n",
+    "    return_type = Numeric\n",
+    "\n",
+    "    def __init__(self, string=None):\n",
+    "        self.string = string\n",
+    "\n",
+    "    def get_function(self):\n",
+    "        def string_count(column):\n",
+    "            assert self.string is not None, \"string to count needs to be defined\"\n",
+    "            # this is a naive implementation used for clarity\n",
+    "            counts = [text.lower().count(self.string) for text in column]\n",
+    "            return counts\n",
+    "\n",
+    "        return string_count"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Now you have a primitive that is reusable for different string values. For example, you can create features based on the number of times the word \"the\" appears in a text. Create an instance of the primitive where the string value is \"the\" and pass the primitive into DFS to generate the features. The feature name will automatically reflect the string value of the primitive."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "es = make_ecommerce_entityset()\n",
+    "\n",
+    "feature_matrix, features = ft.dfs(\n",
+    "    entityset=es,\n",
+    "    target_entity=\"sessions\",\n",
+    "    agg_primitives=[\"sum\", \"mean\", \"std\"],\n",
+    "    trans_primitives=[StringCount(string=\"the\")],\n",
+    ")\n",
+    "\n",
+    "feature_matrix[[\n",
+    "    'STD(log.STRING_COUNT(comments, string=the))',\n",
+    "    'SUM(log.STRING_COUNT(comments, string=the))',\n",
+    "    'MEAN(log.STRING_COUNT(comments, string=the))',\n",
+    "]]"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Features with Multiple Outputs\n",
+    "==============================\n",
+    "\n",
+    "Some calculations can output multiple features. With custom primitives, you can stack and create new features based on more than one output from a single feature.\n",
+    "\n",
+    "\n",
+    "Case Count Example\n",
+    "~~~~~~~~~~~~~~~~~~\n",
+    "\n",
+    "Make a primitive to count the upper case and lower case letters of a text. First, derive a new transform primitive class from :class:`TransformPrimitive`. The primitive will take in a text column as the input and return two numeric columns as the output, so set the input type as :class:`NaturalLanguage`, the return type as :class:`Numeric`, and :code:`number_output_features` to two. Then, override :meth:`get_function` to return the implementation as a function and :meth:`generate_names` to define custom naming for the multiple outputs."
```

Comment:
Same as with above example, could use some more context about the usefulness of the example